tests/View/ViewFactoryTest.php

```diff
@@ -917,14 +917,14 @@ public function testAddingLoops()
         $factory->addLoop([1, 2, 3]);
 
         $expectedLoop = [
-            'iteration' => 0,
+            'iteration' => 1,
```

Comment:
This should not change, why the change?

Maybe this is a big breaking change after all

Reply:
This should function exactly the same. What previously happened is that the code did some initialization with `iteration=0`, after which it immediately incremented the step directly after the `foreach()` struct, thus ending up at `iteration=1` anyway. This extra step is not necessary because you can initialize it correctly immediately.

To address your other comment:

- This PR simplifies the logic thus making the compiled files more readable (compiled view files are often referenced in exceptions so you will end up in these files one way or another)
- It saves a few CPU cycles
- It is mostly old code from 2016 which could use an update